marathon/datadog_checks/marathon/marathon.py

```diff
@@ -50,13 +50,19 @@ class Marathon(AgentCheck):
 
     def __init__(self, name, init_config, instances):
         super(Marathon, self).__init__(name, init_config, instances)
+        timeout = (
+            self.instance.get('timeout')
+            or self.init_config.get('timeout')
+            or self.init_config.get('default_timeout')
+            or self.DEFAULT_TIMEOUT
+        )
         if not ('read_timeout' in self.instance or 'connect_timeout' in self.instance):
             # `default_timeout` config option will be removed with Agent 5
+            self.http.options['timeout'] = timeout
+        else:
             self.http.options['timeout'] = (
-                self.instance.get('timeout')
-                or self.init_config.get('timeout')
-                or self.init_config.get('default_timeout')
-                or self.DEFAULT_TIMEOUT
+                float(self.instance.get('read_timeout', timeout)),
+                float(self.instance.get('connect_timeout', timeout)),
             )
```

Comment:
Code like this one is used in several integrations to support legacy timeout/read_timeout/connect timeout. It's not very pretty and quite hard to maintain.

I think we need to implement something a bit more generic at RequestsWrapper likely to support legacy timeout, by using or extending `HTTP_CONFIG_REMAPPER`.

Reply:
Yeah, probably out of the scope of this PR